### PR TITLE
python3 compatibilty

### PR DESCRIPTION
**Pull request by bastianh**

### CODE REVIEW

File: django_slack/backends.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
+import pprint
+from six.moves import urllib
 from .utils import Backend
 
+
 class Urllib2Backend(Backend):
     def send(self, url, data):
-        r = urllib2.urlopen(urllib2.Request(url, data=urllib.urlencode(data)))
+        r = urllib.urlopen(urllib.Request(url, data=urllib.urlencode(data)))
```
Comment (lamby) [inline django_slack/backends.py]: Please add a comment that this is using ``six.moves`` - otherwise it looks very very weird that a class called "Urllib2Backend" backend is not using ``urllib2``!

File: django_slack/backends.py [hunk 2]
```diff
@@ -22,11 +24,13 @@ def send(self, url, data):
 
         self.validate(r.headers['Content-Type'], r.text)
 
+
```
Comment (lamby) [inline django_slack/backends.py]: please don't make unnecessary whitespace changes in commits

File: django_slack/api.py [hunk 3]
```diff
@@ -32,7 +34,7 @@ def render(component):
     }
 
     # Filter actually defined values
-    data = {k: v for k, v in data.iteritems() if v}
+    data = {k: v for k, v in six.iteritems(data) if v}
```
Comment (lamby) [inline django_slack/api.py]: just change it to items

File: django_slack/api.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import json
 
+import six
```
Comment (lamby) [inline django_slack/api.py]: please group with ``json`` above

**Comment by lamby** [comment]
Very happy to merge once the comments are seen to. :)

File: django_slack/backends.py (hunk 2 above)
Comment (bastianh) [inline django_slack/backends.py]: sorry ... looks like those were added by my editor because of pep8 ( 2 blank lines between classes)

File: django_slack/api.py (hunk 4 above)
Comment (bastianh) [inline django_slack/api.py]: no problem, but the blank line was added intentionally because of pep-8 rules for imports:
```Imports should be grouped in the following order:

standard library imports
related third party imports
local application/library specific imports
You should put a blank line between each group of imports.```

Comment (lamby) [inline django_slack/api.py]: I have no real problem with PEP8 etc. but you should not mix functionality and cosmetic changes, regardless of how strong the reason or whether "it was my editor".

Feel free to make a separate PR - I would just want to see only python3 changes.

**Comment by bastianh** [comment]
best would be to do a new pull request from the current master I guess.

**Comment by lamby** [comment]
Go for it :)

**Comment by lamby** [comment]
(Just checking you're not waiting on me for this? As in, the "cosmetic" commit here actually reverts changes made by the other one, etc. etc.)

**Comment by lamby** [comment]
Oh, I forgot about this - how are you getting on?

**Comment by Eraldo** [comment]
The pull request looks good to me.
Can this be merged?

**Comment by lamby** [comment]
Hm? I think I already mention here it is not ready to merge :)

**Comment by Eraldo** [comment]
What is missing?

**Comment by lamby** [comment]
Sorry, do you not see my comments? They should be pretty self-explanatory.

**Comment by Eraldo** [comment]
I do.. but I am afraid I do not understand what is wrong. :|

**Comment by lamby** [comment]
> As far as I can tell there are comments.

Right :) And I'm waiting for those issues to be resolved before merging. Just merging something because it "works" is rarely a good idea!

**Comment by Eraldo** [comment]
Thanks for clarifying.. I misunderstood what comments you were talking about.
You meant source code comments.. I thought you were talking about this thread conversation.
Okay.. I'm on it. ;)

**Comment by lamby** [comment]
I've reworked this to a branch https://github.com/lamby/django-slack/tree/python3-support which fixes some issues with the urllib backend.

Can you test it please @Eraldo? I don't have a convenient py3 setup to hand right this second.

**Comment by Eraldo** [comment]
Sure. I will test it now. :)

**Comment by lamby** [comment]
Fixed on master,
